cdc/sink/ddlsink/mysql/async_ddl.go

```diff
@@ -16,24 +16,27 @@ package mysql
 import (
 	"context"
 	"database/sql"
+	"errors"
 	"fmt"
 	"time"
 
 	"github.com/pingcap/log"
 	timodel "github.com/pingcap/tidb/pkg/parser/model"
 	"github.com/pingcap/tiflow/cdc/model"
-	"github.com/pingcap/tiflow/pkg/errors"
 	"go.uber.org/zap"
 )
 
+// TODO: Use the flollowing SQL to check the ddl job status after tidb optimize
```

Comment:
Did you check the performance of this new query? I take a glimpse into ADMIN SHOW DDL JOBS and it still loads many ddl job before filtering

Reply:
We have tested scenarios with a large number of historical DDLs:
![image](https://github.com/user-attachments/assets/a2bbea88-b2eb-44f1-8f97-f70ad0e0423f)

Also, note that the tidb implementation here [queries all records](https://github.com/pingcap/tidb/blob/master/pkg/executor/executor.go#L417) in the `tidb_ddl_job` table, and query performance depends on the size of the `tidb_ddl_job`. Seems the results of this query have to be stored in memory. Maybe it is more appropriate to push down the filter criteria in tidb?

In this PR, we add a timeout to control the max query cost.